eng/pipelines/templates/jobs/ci.yml

```diff
@@ -87,6 +87,7 @@ jobs:
           LogFilePath: $(Build.ArtifactStagingDirectory)/pack.binlog
       - script: >-
           dotnet pack eng/service.proj -warnaserror
+          /p:ValidateRunApiCompatSetting=true
```

Comment:
Should we move this to the release pipeline YAML rather than enforce on every run?

Reply:
Hmm I would rather have the code fail to even merge into the repo rather than waiting for release day.